plugins/core-plugin/src/main/resources/Dockerfile-template-yaml

```diff
@@ -1,16 +1,68 @@
-FROM ${baseContainerImage}
+#===================================================================#
+# Create build environment from base Python template launcher image #
+#===================================================================#
+FROM gcr.io/dataflow-templates-base/python311-template-launcher-base:latest as python-base
 
-# Copy template files to /template
+# Build args
 ARG WORKDIR=/template
-COPY main.py /template
-COPY requirements.txt /template
-COPY ${yamlTemplateName} /template/template.yaml
+ARG REQUIREMENTS_FILE=requirements.txt
+ARG BEAM_VERSION=${beamVersion}
+ARG BEAM_PACKAGE=apache-beam[dataframe,gcp,test,yaml]==$BEAM_VERSION
+ARG PY_VERSION=${pythonVersion}
+
+# Copy template files to /template
+RUN mkdir -p $WORKDIR
+COPY main.py requirements.txt* /template/
+<#if copyOtherFiles??>COPY ${copyOtherFiles} /template/</#if>
```

Comment:
Why did we switch to this if block? If `copyOtherFiles` is empty shouldn't `COPY main.py requirements.txt* /template/` be valid?

Reply:
I'm also not familiar with this `<#if` syntax - is this a valid dockerfile? It looks more like terraform if statements.

Reply:
Discussed offline, this is resolved by freemarker (this is a txt file)